Config updates

Tiagoquix opened the pull request.

Tiagoquix committed:
comfig.cfg updates

Tiagoquix committed:
modules.md updates

treacherousfiend reviewed (commented):
With all due respect (alongside the fact that this is a draft), some of these changes seem to be preference things more than anything, specifically:

`//cl_crosshair_file crosshair3`
`//engine_no_focus_sleep 50`
`//mat_bufferprimitives 1`

and changing `sourcetv_off` and `sourcetv_on` with `sourcetv_compatibility_off` and `sourcetv_compatibility_on`

Also not entirely sure why you added `fps_max 300` to `fpscap_powersaver` as the TF2 fps cap can add some kind of bad overhead.

I do approve of commenting out all of the `tf_hud` cvars, as well as `cl_mute_all_comms` as those are things that I don't see any reason for comfig to change, and should only be changed by the user.

Tiagoquix commented:
> With all due respect (alongside the fact that this is a draft), some of these changes seem to be preference things more than anything, specifically:
> 
> `//cl_crosshair_file crosshair3`
> `//engine_no_focus_sleep 50`
> `//mat_bufferprimitives 1`
> 
> and changing `sourcetv_off` and `sourcetv_on` with `sourcetv_compatibility_off` and `sourcetv_compatibility_on`
> 
> Also not entirely sure why you added `fps_max 300` to `fpscap_powersaver` as the TF2 fps cap can add some kind of bad overhead.
> 
> I do approve of commenting out all of the `tf_hud` cvars, as well as `cl_mute_all_comms` as those are things that I don't see any reason for comfig to change, and should only be changed by the user.

I did put `crosshair3` on `cl_crosshair_file` just to it don't be empty. This does not change anything, since the comfig while packaged will always have the comments removed.

About `engine_no_focus_sleep`, its default value is `50`, and `mat_bufferprimitives`, its default value is `1`.

`sourcetv_off and _on to sourcetv_compatibility` I guess that somewhere in the future Valve may use these, so I added `_compatibility` to them, but I can remove it, no problems.

About `tf_` cvars and `cl_mute_all_comms`, I commented them out because the config was forcing the default values for them.

Tiagoquix committed:
force voice_steal 2 on comfig.cfg too

treacherousfiend commented:
Ah, I think you might be in the wrong place, valve isn't going to use any of this, you might be thinking of team comtress 2, which is a similar project which is trying to optimize the source code of the game, which eventually _will_ be sent to valve. 
Regarding everything else, thanks for clarifying, I don't have any qualms with this PR now that I know its just removing default values

Tiagoquix committed:
modules_run.cfg updates

micwoj92 commented:
> `sourcetv_off and _on to sourcetv_compatibility` I guess that somewhere in the future Valve may use these, so I added `_compatibility` to them, but I can remove it, no problems.


No matter how hard I try but I do not understand what do you mean by this part

Tiagoquix committed:
bug_report.md updates

melvyn2 commented:
> > `sourcetv_off and _on to sourcetv_compatibility` I guess that somewhere in the future Valve may use these, so I added `_compatibility` to them, but I can remove it, no problems.
> 
> No matter how hard I try but I do not understand what do you mean by this part

Valve may create the `sourcetv_on/_off` aliases in the future.

micwoj92 commented:
> > > `sourcetv_off and _on to sourcetv_compatibility` I guess that somewhere in the future Valve may use these, so I added `_compatibility` to them, but I can remove it, no problems.
> > 
> > 
> > No matter how hard I try but I do not understand what do you mean by this part
> 
> Valve may create the `sourcetv_on/_off` aliases in the future.

How do you even come to this thought?

Tiagoquix commented:
> > `sourcetv_off and _on to sourcetv_compatibility` I guess that somewhere in the future Valve may use these, so I added `_compatibility` to them, but I can remove it, no problems.
> 
> No matter how hard I try but I do not understand what do you mean by this part

Maybe Valve would use `sourcetv_off` and `sourcetv_on` on the future for something. But thinking about it, doesn't seems to be logically correctly, since Valve miserably updates TF2. I will remove `_compatibility` from the SourceTV module.

Tiagoquix committed:
remove _compatibility (comfig.cfg)

Tiagoquix committed:
remove _compatibility (modules_run.cfg)

treacherousfiend commented:
the modules are not in the game at all, they are aliases added by comfig for easy editing so people don't need to look through cvars all the time

Tiagoquix committed:
remove _compatibility (modules.md)

Tiagoquix commented:
> the modules are not in the game at all, they are aliases added by comfig for easy editing so people don't need to look through cvars all the time

I know but I think Valve would add the "sourcetv" command in the future or something like. But you shouldn't worry about this anymore, I have reverted the changes.

micwoj92 commented on docs/customization/modules.md:
```diff
@@ -454,19 +454,20 @@ Setting it higher can reduce input delay.
+* **`fpscap=240`**: Sets FPS cap to 240 FPS.
+* **`fpscap=300`**: Sets FPS cap to 300 FPS.
+* **`fpscap=360`**: Sets FPS cap to 360 FPS.
+* **`fpscap=400`**: Sets FPS cap to 400 FPS.
```
needs update to modules template

Tiagoquix replied:
What?

treacherousfiend replied:
I see, you forgot to add `fpscap=400` to the modules page

micwoj92 replied:
I opened pr on your repo

micwoj92 committed:
Update modules.cfg

mastercoms commented:
Some of these have different defaults on Linux/mac.

Tiagoquix committed:
Merge pull request #1 from micwoj92/patch-4

Update modules.cfg

Tiagoquix committed:
modules.md outlines module

Tiagoquix committed:
bring back mat_bufferprimitves and engine_no_focus_sleep

Tiagoquix committed:
Update comfig.cfg

Tiagoquix committed:
Update comfig.cfg

Tiagoquix committed:
Update modules.md

Tiagoquix committed:
Update very-low.cfg

Tiagoquix committed:
Update comfig.cfg

Tiagoquix committed:
Update README.md

Tiagoquix committed:
Update index.md

Tiagoquix committed:
Update README.md

Tiagoquix committed:
Update troubleshoot.md

Tiagoquix committed:
Update faq.md

Tiagoquix committed:
Update modules.md

Tiagoquix committed:
Update modules.md

Tiagoquix committed:
Merge branch 'develop' into patch-1

Tiagoquix committed:
Update modules.json

Tiagoquix committed:
Update preset_modules.json

Tiagoquix commented:
Summary of changes I made:

- Changed GitHub's bug report template to include the explanation between the parentheses;
- Changed `README.MD` and `docs/index.md` just for pure perfectionism;
- Added 360 FPS Cap back since there is 360Hz monitors. The 400 FPS Cap is just to match the CS:GO default (`fpscap=400` should be the default to improve load times);
- Forced `fps_max` to be `300` on `fpscap_powersaver` just to enforce its default value.
- Removed some cvars being forced to their default values (//);
- Changed `r_spray_lifetime` to `2` instead of `1` on `sprays_on`;
- Added `mat_bloom_scalefactor_scalar 1` to `post_processing_off` and `post_processing_low`;
- Removed some unnecessary zeros in the levels of the post-processing module;
- Removed one `r_dopixelvisibility` explanation, as it was commented two times;
- Added `mat_supportflashlight 0` to skip the load of flashlight shader combos;
- Added a commentary to `cheap_water_never`;
- Forced `mod_forcetouchdata` to `1` by default;
- Forced `voice_steal 2` on `comfig.cfg` too (since it is in `game_overrides.cfg`);
- Cleaned up unnecessary spaces on the logo module in `comfig.cfg`;
- Revised `modules.md` and made some tweaks;
- Changed outline module levels;
- Added more information on the docs' FAQ.
- Removed some redundances on the docs' troubleshooting page;
- Changed commentaries in the very-low preset;
- Corrected killstreak levels in `data/preset_modules.json`.

@mastercoms

mastercoms commented:
I took some of these changes manually in d239af670d92f0b67558a1a77c7810668b959cc0.

Some of the formatting/grammar changes were not correct for what I wanted.

Power saving mode sets fps_max. So setting fps_max would just reverse the power saving mode.

mat_supportflashlight is already disable automatically in TF on engine startup.

mat_bloom_scalefactor_scalar is not used on those modes, as bloom is disabled.

mod_forcetouchdata is not archived and not changed anywhere in the source code, so forcing it is useless.

voice_steal 2 is useless to set at game start up, as it is set to 1 on map loads.

Outline modules are intentionally like that to give access to certain features which have been requested by users.

FAQ page does not need that information about classes -- it's already on the custom configs page.

The cvars you did comment are archive cvars which are being forced to a value, or are intentional changes.

Tiagoquix commented:
> Outline modules are intentionally like that to give access to certain features which have been requested by users.

Yeah but the low level enables more outlines than the medium level. I'll do a separeted PR for the outline module.

> The cvars you did comment are archive cvars which are being forced to a value, or are intentional changes.

Yes but they are being forced to their defaults. Why?
